src/python/pants/backend/jvm/tasks/jvm_compile/jvm_classpath_publisher.py

```diff
@@ -29,7 +35,11 @@ def _output_folder(self):
   def execute(self):
     basedir = os.path.join(self.get_options().pants_distdir, self._output_folder)
     runtime_classpath = self.context.products.get_data('runtime_classpath')
+
     targets = self.context.targets()
+    if self.get_options().transitive_only:
+      targets = list(OrderedSet(targets) - set(self.context.target_roots))
```

Comment:
Does `OrderedSet` - `set` gives back an `OrderedSet`?

Also
```
targets = list(OrderedSet(targets) - set(self.context.target_roots)) if self.get_options().transitive_only else self.context.targets()
```
would probably be cleaner logic wise.

Reply:
> Does `OrderedSet` - `set` gives back an `OrderedSet`?

I wasn't sure before, but after checking, I believe so:
```
X> pex twitter.common.collections
/var/folders/_r/gknsp2dd6bs3k1y9s6btxbyc0000gn/T/tmpidggo1ny/.bootstrap/pex/environment.py:330: PEXWarning: The `pkg_resources` package was loaded from a pex vendored version when declaring namespace packages defined by twitter.common.collections 0.3.11. The twitter.common.collections 0.3.11 distribution should fix its `install_requires` to include `setuptools`
  .format(dist=dist))
Python 3.7.4 (default, Oct 12 2019, 18:55:28)
[Clang 11.0.0 (clang-1100.0.33.8)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from twitter.common.collections import OrderedSet
>>> OrderedSet(['a', 'b', 'd']) - set('b')
OrderedSet(['a', 'd'])
```

> `targets = list(OrderedSet(targets) - set(self.context.target_roots)) if self.get_options().transitive_only else self.context.targets()`

I think you might be thinking of:
```python
targets = list(OrderedSet(self.get_targets()) - set(self.context.target_roots) if self.get_options().transitive_only else self.get_targets()`
```
Having to write out `self.get_targets()` or `self.context.targets()` twice in the same line feels like a lot of text to write, personally. I think I agree with your idea but I can't see a way to write it that wouldn't go over 100 characters right now.

Reply:
actually if `targets` is a `OrderedSet`, does it still need to be a list?

Reply:
re formatting, having `\` is probably fine
```
    targets = OrderedSet(self.get_targets()) - set(self.context.target_roots) \
      if self.get_options().transitive_only else self.get_targets()
```

Reply:
oh hm, that's great! I'll do that! Thank you!